openedx_learning/core/components/models.py

```diff
@@ -28,6 +28,45 @@
 from ..publishing.models import LearningPackage
 
 
+class ComponentType(models.Model):
+    """
+    Normalized representation of a type of Component.
+
+    The only namespace being used initially will be 'xblock.v1', but we will
+    probably add a few others over time, such as a component type to represent
+    packages of files for things like Files and Uploads or python_lib.zip files.
+
+    Make a ForeignKey against this table if you have to set policy based on the
+    type of Components–e.g. marking certain types of XBlocks as approved vs.
+    experimental for use in libraries.
+    """
+    id = models.AutoField(primary_key=True)
+
+    # namespace and name work together to help figure out what Component needs
+    # to handle this data. A namespace is *required*. The namespace for XBlocks
+    # is "xblock.v1" (to match the setup.py entrypoint naming scheme).
+    namespace = case_sensitive_char_field(max_length=100, blank=False)
+
+    # name is a way to help sub-divide namespace if that's convenient. This
+    # field cannot be null, but it can be blank if it's not necessary. For an
+    # XBlock, this corresponds to tag, e.g. "video". It's also the block_type in
+    # the UsageKey.
+    name = case_sensitive_char_field(max_length=100, blank=True)
```

Comment:
Should we set `null=True`, so that we can't have multiple ComponentTypes with the same namespace and and empty names?

Reply:
I'm not sure if I understood the question correctly. The way the code is now (with `null=False`), the uniqueness constraint on `(namespace, name)` means that there can't be multiple entries for the same namespace with blank names. But turning `null=True` would allow that to happen, since nulls don't evaluate as equal to each other, so the uniqueness constraint doesn't help.

Reply:
Sorry, I misread https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.Field.null. You're right--this looks good.